refactor(trading): middleware loading data to independent thunk

adderpositive opened the pull request:
## Description
- refactored middleware loading data to independent thunk
- renamed `trading` to `tradingNew` as temporary solution 

## Related Issue
Partially resolve https://github.com/trezor/trezor-suite/issues/17049

jbazant commented on suite-common/trading/src/middlewares/tradingMiddleware.ts:
```diff
@@ -1,69 +1,5 @@
-
-        return next(action);
-    },
+export const prepareTradingMiddleware = createMiddlewareWithExtraDeps((action, { next }) =>
+    next(action),
 );
```
question: Why to do we need this?

adderpositive replied:
good question :) dead code, can be deleted for now.

adderpositive replied:
Updated

jbazant reviewed (commented):
I have one question. Rest seem OK.

adderpositive commented:
@jbazant I have added one more commit https://github.com/trezor/trezor-suite/pull/17106/commits/b7e9b2e099897a721cedaa8a2b537aa0cf925c6a. I have renamed `trading` to `tradingNew` to avoid conflict with the current reducer on the desktop. It is a temporary solution.

jbazant reviewed (approved):
LGTM!

tomasklim commented on suite-common/trading/src/hooks/useTradingInfo.ts:
```diff
@@ -66,9 +67,34 @@ const sortPopularCurrencies = (
+    const tradingInfo = useSelector(selectTradingInfoLegacy);
+    // const tradingNewInfo = useSelector(selectTradingInfo);
+
+    const getInfo = () =>
+        /*
+        if (section) {
+            // TODO: trading - refactor only buy is refactored for now
+            return section === 'buy' ? tradingNewInfo : tradingInfo;
+        }
+        */
+
+        tradingInfo;
```
this is really ugly 😀

adderpositive replied:
Updated here (next PR) - https://github.com/trezor/trezor-suite/pull/17175/files#diff-24ea559bff1242d0505b46e742a727d740e152ccf68476c111abbfd295407a42R77

tomasklim commented on suite-common/trading/src/hooks/useTradingInfo.ts:
```diff
@@ -66,9 +67,34 @@ const sortPopularCurrencies = (
+    const { platforms, coins } = useMemo(
+        () => ({
+            platforms: info?.platforms ?? {},
+            coins: info?.coins ?? {},
```
in some future update, can you rename coins to tokens? or does it come like that from API?

adderpositive replied:
Yes, it comes from API in the same form.
